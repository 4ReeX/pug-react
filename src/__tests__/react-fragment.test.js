import testHelper from './test-helper';

describe('with each', () => {
  testHelper(__dirname + '/react-fragment-each.input.js');
});

describe('with if', () => {
  testHelper(__dirname + '/react-fragment-if.input.js');
});

describe('with inline javascript', () => {
  testHelper(__dirname + '/react-fragment-inline-js.input.js');
});

describe('basic', () => {
  testHelper(__dirname + '/react-fragment.input.js');
});
