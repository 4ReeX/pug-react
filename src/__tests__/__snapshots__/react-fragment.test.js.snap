// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic JavaScript output: transformed source code 1`] = `
"module.exports = (
  <React.Fragment>
    <div />
    <div />
  </React.Fragment>
);
"
`;

exports[`basic html output: generated html 1`] = `
Array [
  <div />,
  <div />,
]
`;

exports[`basic static html output: static html 1`] = `"<div></div><div></div>"`;

exports[`with each JavaScript output: transformed source code 1`] = `
"module.exports = (
  <React.Fragment>
    {[1, 2, 3].map(item => {
      return <div key={item}>{item}</div>;
    })}
    <div>Next item</div>
  </React.Fragment>
);
"
`;

exports[`with each html output: generated html 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
  <div>
    Next item
  </div>,
]
`;

exports[`with each static html output: static html 1`] = `"<div>1</div><div>2</div><div>3</div><div>Next item</div>"`;

exports[`with if JavaScript output: transformed source code 1`] = `
"module.exports = (
  <React.Fragment>
    {true ? <div>Truthy</div> : null}
    <div>Next item</div>
  </React.Fragment>
);
"
`;

exports[`with if html output: generated html 1`] = `
Array [
  <div>
    Truthy
  </div>,
  <div>
    Next item
  </div>,
]
`;

exports[`with if static html output: static html 1`] = `"<div>Truthy</div><div>Next item</div>"`;

exports[`with inline javascript JavaScript output: transformed source code 1`] = `
"let _i;

module.exports = (
  <React.Fragment>
    {((_i = 100), null)}
    <div>{_i}</div>
  </React.Fragment>
);
"
`;

exports[`with inline javascript html output: generated html 1`] = `
<div>
  100
</div>
`;

exports[`with inline javascript static html output: static html 1`] = `"<div>100</div>"`;
